src/clarity_cli.rs

```diff
@@ -1526,14 +1618,36 @@ pub fn invoke_command(invoked_by: &str, args: &[String]) -> (i32, Option<serde_j
 
             let (_, _, result_and_cost) = in_block(header_db, marf_kv, |header_db, mut marf| {
                 let result_and_cost = with_env_costs(mainnet, &header_db, &mut marf, |vm_env| {
-                    vm_env.execute_transaction(sender, contract_identifier, &tx_name, &arguments)
+                    if coverage_folder.is_some() {
+                        vm_env.set_coverage_reporter(CoverageReporter::new());
+                    }
+                    (
+                        vm_env.execute_transaction(
+                            sender,
+                            contract_identifier,
+                            &tx_name,
+                            &arguments,
+                        ),
+                        vm_env.take_coverage_reporter(),
+                    )
                 });
-                (header_db, marf, result_and_cost)
+                let ((result, coverage), cost) = result_and_cost;
+                (header_db, marf, (result, cost, coverage))
             });
 
             match result_and_cost {
-                (Ok((x, asset_map, events)), cost) => {
+                (Ok((x, asset_map, events)), cost, coverage) => {
                     if let Value::Response(data) = x {
+                        if let Some(ref coverage_folder) = coverage_folder {
+                            let mut coverage_file = PathBuf::from(coverage_folder);
+                            coverage_file.push(&format!("execute_{}", get_epoch_time_ms()));
+                            coverage_file.set_extension("clarcov");
+
+                            coverage
+                                .expect("Failed to recover coverage reporter when coverage was requested")
+                                .to_file(&coverage_file)
+                                .expect("Coverage reference file generation failure");
+                        }
```

Comment:
this could be broken up into a separate function since this fragment is repeated in multiple places - your call though